packages/geocoder/src/geocoders/abstract-geocoder.ts

```diff
@@ -0,0 +1,231 @@
+import { fromCoordinates } from "@conveyal/lonlat";
+
+// Prettier does not support typescript annotation
+// eslint-disable-next-line prettier/prettier
+import type { LonLatInput } from "@conveyal/lonlat"
+import type { Feature, FeatureCollection } from "geojson"
+
+type Rect = {
+  maxLat: number;
+  maxLon: number;
+  minLat: number;
+  minLon: number;
+};
+
+type Boundary = {
+  country?: string;
+  rect?: Rect;
+};
+
+type GeocoderAPI = {
+  autocomplete: (query: AutocompleteQuery) => Promise<unknown>;
+  reverse: (query: ReverseQuery) => Promise<unknown>;
+  search: (query: SearchQuery) => Promise<unknown>;
+};
+
+// These settings get added to the request to the geocoder
+export type GeocoderConfig = {
+  apiKey?: string;
+  baseUrl?: string;
+  boundary?: Boundary;
+  focusPoint?: LonLatInput;
+  options?: RequestInit;
+  sources?: string;
+  size?: number;
+  reverseUseFeatureCollection?: boolean;
+};
+
+export type ReverseQuery = {
+  apiKey?: string;
+  format?: boolean;
+  options?: RequestInit;
+  point?: LonLatInput;
+  url?: string;
+  sources?: string;
+};
+
+export type AutocompleteQuery = {
+  apiKey?: string;
+  boundary?: Boundary;
+  focusPoint?: LonLatInput;
+  format?: boolean;
+  options?: RequestInit;
+  size?: number;
+  text?: string;
+  url?: string;
+  sources?: string;
+};
+
+export type SearchQuery = {
+  apiKey?: string;
+  boundary?: Boundary;
+  focusPoint?: LonLatInput;
+  format?: boolean;
+  options?: RequestInit;
+  size?: number;
+  text?: string;
+  url?: string;
+  sources?: string;
+};
+
+export type MultiGeocoderResponse = FeatureCollection & {
+  isomorphicMapzenSearchQuery?: SearchQuery & AutocompleteQuery & ReverseQuery;
+};
+
+export type SingleGeocoderResponse = {
+  isomorphicMapzenSearchQuery?: SearchQuery & AutocompleteQuery & ReverseQuery;
+  lat: number;
+  lon: number;
+  name: string;
+  rawGeocodedFeature: Feature;
+};
+/**
+ * Create customized geocoder functions given a certain geocoding API, the
+ * config for the geocoder and response rewrite functions specific to this
+ * application. Any geocoder API that is added is expected to have an API that
+ * behaves very closely to https://github.com/conveyal/isomorphic-mapzen-search
+ */
+export default class Geocoder {
+  geocoderConfig: GeocoderConfig;
+
+  api: GeocoderAPI;
+
+  constructor(geocoderApi?: GeocoderAPI, geocoderConfig?: GeocoderConfig) {
+    this.api = geocoderApi;
+    this.geocoderConfig = geocoderConfig;
+    console.log(this.geocoderConfig);
```

Comment:
Uh oh I think this snuck in by accident!

Reply:
Fixed in a3cc58b